src/traces/pointcloud/convert.js

```diff
@@ -0,0 +1,218 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var createPointCloudRenderer = require('gl-pointcloud2d');
+
+var str2RGBArray = require('../../lib/str2rgbarray');
+var getTraceColor = require('../scatter/get_trace_color');
+
+var AXES = ['xaxis', 'yaxis'];
+
+function Pointcloud(scene, uid) {
+    this.scene = scene;
+    this.uid = uid;
+
+    this.pickXData = [];
+    this.pickYData = [];
+    this.xData = [];
+    this.yData = [];
+    this.textLabels = [];
+    this.color = 'rgb(0, 0, 0)';
+    this.name = '';
+    this.hoverinfo = 'all';
+
+    this.idToIndex = new Int32Array(0);
+    this.bounds = [0, 0, 0, 0];
+
+    this.pointcloudOptions = {
+        positions: new Float32Array(0),
+        idToIndex: this.idToIndex,
+        sizemin: 0.5,
+        sizemax: 12,
+        color: [0, 0, 0, 1],
+        areaRatio: 1,
+        borderColor: [0, 0, 0, 1]
+    };
+    this.pointcloud = createPointCloudRenderer(scene.glplot, this.pointcloudOptions);
+    this.pointcloud._trace = this; // scene2d requires this prop
+}
+
+var proto = Pointcloud.prototype;
+
+proto.handlePick = function(pickResult) {
+
+    var index = this.idToIndex[pickResult.pointId];
+
+    // prefer the readout from XY, if present
+    return {
+        trace: this,
+        dataCoord: pickResult.dataCoord,
+        traceCoord: this.pickXYData ?
+            [this.pickXYData[index * 2], this.pickXYData[index * 2 + 1]] :
+            [this.pickXData[index], this.pickYData[index]],
+        textLabel: Array.isArray(this.textLabels) ?
+            this.textLabels[index] :
+            this.textLabels,
+        color: this.color,
+        name: this.name,
+        hoverinfo: this.hoverinfo
+    };
+};
+
+proto.update = function(options) {
+
+    this.textLabels = options.text;
+    this.name = options.name;
+    this.hoverinfo = options.hoverinfo;
+    this.bounds = [Infinity, Infinity, -Infinity, -Infinity];
+
+    this.updateFast(options);
+
+    this.color = getTraceColor(options, {});
+};
+
+proto.updateFast = function(options) {
+    var x = this.xData = this.pickXData = options.x;
+    var y = this.yData = this.pickYData = options.y;
+    var xy = this.pickXYData = options.xy;
+
+    var userBounds = options.xbounds && options.ybounds;
+    var index = options.indices;
+
+    var len,
+        idToIndex,
+        positions,
+        bounds = this.bounds;
+
+    var xx, yy, i;
+
+    if(xy) {
+
+        positions = xy;
+        len = xy.length >>> 1;
```

Comment:
bitwise ops are cool :sunglasses: but can you add a short comment above describing what this does (for folks unfamiliar with them).